lib/puppet/ssl/host.rb

```diff
@@ -372,6 +374,47 @@ def state
 
   private
 
+  # Ensures that CRL is either on disk, or that CRL checking has been disabled
+  def ensure_crl_if_needed
+    if crl_usage && !Puppet::FileSystem.exist?(crl_path)
+      # The CertificateRevocationList indirector will attempt to download the
+      # CRL from the CA if it does not exist on disk.
+      old_crl_usage = crl_usage
+      @crl_usage = false
```

Comment:
Yeah, my first pass was to basically inline the behavior of Context#override, which afiact isn't thread safe either.

I was thinking that in PUP-8654, when we use @Magisus 's code to download the CRL we could hopefully configure the connection we use to explicitly not use CRL checking, rather than the double call to Host#ssl_store that we're doing today (iow, I'm hoping we don't need this state juggling when we don't have to use the indirector).